__tests__/__unit__/extension.unit.test.ts

```diff
@@ -1375,907 +1494,769 @@ describe("Extension Unit Tests", () => {
+    it("Testing that that openPS credentials prompt ends in error", async () => {
+        showTextDocument.mockReset();
+        openTextDocument.mockReset();
+        showQuickPick.mockReset();
+        showInputBox.mockReset();
         const sessionwocred = new imperative.Session({
             user: "",
             password: "",
             hostname: "fake",
+            port: 443,
             protocol: "https",
             type: "basic",
         });
-        // testing if no session is defined (can happen while saving from favorites)
-        const nodeWitoutSession = new ZoweDatasetNode("HLQ.TEST.AFILE", vscode.TreeItemCollapsibleState.None,
-                                                        null, null, undefined, undefined, profileOne);
-        testTree.getChildren.mockReturnValueOnce([nodeWitoutSession]);
-        concatChildNodes.mockReturnValueOnce([nodeWitoutSession]);
-        const getSessionSpy = jest.spyOn(mvsApi, "getSession").mockReturnValueOnce(sessionwocred);
-        await dsActions.saveFile(testDoc0, testTree);
-        // tslint:disable-next-line: no-magic-numbers
-        expect(getSessionSpy.mock.calls.length).toBe(3);
-        expect(getSessionSpy.mock.results[0].value).toEqual(sessionwocred);
-
-        // testing if no documentSession is found (no session + no profile)
+        const dsNode = new ZoweDatasetNode("testSess", vscode.TreeItemCollapsibleState.Expanded, sessNode, sessionwocred);
+        dsNode.contextValue = globals.DS_SESSION_CONTEXT;
         Object.defineProperty(Profiles, "getInstance", {
             value: jest.fn(() => {
                 return {
                     allProfiles: [{name: "firstName", profile: {user:undefined, password: undefined}}, {name: "secondName"}],
-                    loadNamedProfile: () => undefined
+                    defaultProfile: {name: "firstName"},
+                    validProfile: ValidProfileEnum.INVALID,
+                    checkCurrentProfile: jest.fn(),
+                    loadNamedProfile: mockLoadNamedProfile
                 };
             })
         });
-        createBasicZosmfSession.mockReset();
-        testTree.getChildren.mockReset();
+
+        await dsActions.openPS(dsNode, true, testTree);
+        expect(Profiles.getInstance().validProfile).toBe(ValidProfileEnum.INVALID);
+        showQuickPick.mockReset();
+        showInputBox.mockReset();
+        showInformationMessage.mockReset();
         showErrorMessage.mockReset();
-        testTree.getChildren.mockReturnValueOnce([nodeWitoutSession]);
-        await dsActions.saveFile(testDoc0, testTree);
-        expect(showErrorMessage.mock.calls.length).toBe(1);
-        expect(showErrorMessage.mock.calls[0][0]).toBe("Couldn't locate session when saving data set!");
+    });
 
-        testTree.getChildren.mockReset();
-        createBasicZosmfSession.mockReset();
+    it("Testing that that openPS credentials with favorites ends in error", async () => {
```

Comment:
All credential checks have been covered in profiles.unit.test.ts so they can be removed